hbase-client/src/main/java/org/apache/hadoop/hbase/security/SaslWrapHandler.java

```diff
@@ -18,80 +18,31 @@
 package org.apache.hadoop.hbase.security;
 
 import javax.security.sasl.SaslClient;
-import org.apache.hadoop.hbase.exceptions.ConnectionClosedException;
 import org.apache.yetus.audience.InterfaceAudience;
 
 import org.apache.hbase.thirdparty.io.netty.buffer.ByteBuf;
-import org.apache.hbase.thirdparty.io.netty.buffer.Unpooled;
 import org.apache.hbase.thirdparty.io.netty.channel.ChannelHandlerContext;
-import org.apache.hbase.thirdparty.io.netty.channel.ChannelOutboundHandlerAdapter;
-import org.apache.hbase.thirdparty.io.netty.channel.ChannelPromise;
-import org.apache.hbase.thirdparty.io.netty.channel.CoalescingBufferQueue;
-import org.apache.hbase.thirdparty.io.netty.util.ReferenceCountUtil;
-import org.apache.hbase.thirdparty.io.netty.util.concurrent.PromiseCombiner;
+import org.apache.hbase.thirdparty.io.netty.handler.codec.MessageToByteEncoder;
 
 /**
  * wrap sasl messages.
  */
 @InterfaceAudience.Private
-public class SaslWrapHandler extends ChannelOutboundHandlerAdapter {
+public class SaslWrapHandler extends MessageToByteEncoder<ByteBuf> {
```

Comment:
Nice, I think extending `MessageToByteEncoder` seems to be making it much simpler. Will try deploying on a secure cluster.